### DISCUSSION TITLE
Avoid adding quotes to strings in extensions

### DISCUSSION THREAD

**Discussion by artelk** [linq2db/linq2db#3121] [Q&A] [answered]
Can I avoid wrapping the name into additional single quotes for my extension
```cs
[Sql.Extension("Something(\"{name}\")")]
static Foo Bar([ExprParameter] string name) {...}
```
Calling `Bar("field")` produces `Something("'field'")` (wrapped to additional `'` quotes).

**Comment by sdanyliv** [comment, marked as answer]
Check Collation realisation: https://github.com/linq2db/linq2db/blob/master/Source/LinqToDB/Sql/Sql.Collate.cs#L23

**Reply by artelk** [reply]
Thanks. Actually I also have additional values param which is an array of some expressions.
```cs
[Sql.Extension("XmlElement(\"{name}\",  {values, ', '})")]
static Foo XmlElement([ExprParameter] string name, [ExprParameter] params object[] values) {...}
```
How can I handle the `values` with my custom IExtensionCallBuilder implementation?

**Reply by artelk** [reply]
Found https://github.com/linq2db/linq2db/blob/master/Source/LinqToDB/Sql/Sql.Expressions.cs#L601
With a few hacks I was finally able to handle the `values` 😄

**Reply by sdanyliv** [reply]
Well if it is useful extension, do not forget to share. Also it will be good for regression testing.

**Reply by artelk** [reply]
https://gist.github.com/artelk/1ff5c291beb888dcb1a3854d3c0da093
See XmlElementBuilder, 'paramNumbers' string and the created SqlExpression.
